File: java/src/hackerrank/GetMoneySpent.java
package hackerrank;

import static org.junit.jupiter.api.Assertions.assertEquals;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;

// https://www.hackerrank.com/challenges/electronics-shop/problem
public class GetMoneySpent {

	public static int getMoneySpent(int[] keyboards, int[] drives, int b) {
		List<Integer> results = new ArrayList<>();
		for (int i = 0; i < keyboards.length; i++) {
			for (int j = 0; j < drives.length; j++) {
				int sum = keyboards[i] + drives[j];
				if (b >= sum) results.add(sum);
			}
		}
		return results.isEmpty() ? -1 : Collections.max(results);
	}


	@Test
	@DisplayName("Test sample 0")
	void testGetMoneySpent0() {
		int[] keyboards0 = {3,1}, drives0 = {5,2,8}; 
		int b = 10, expected = 9;
		int result = getMoneySpent(keyboards0, drives0, b);
		assertEquals(expected, result);
	}
	
	@Test
	@DisplayName("Test sample 1")
	void testGetMoneySpent() {
		int[] keyboards1 = {4}, drives1 = {5}; 
		int b = 5, expected = -1;
		int result = getMoneySpent(keyboards1, drives1, b);
		assertEquals(expected, result);
	}
	
}

File: kotlin/src/hackerrank/getMoneySpent.kt
package hackerrank

import org.junit.jupiter.api.Test
import kotlin.test.assertEquals

// https://www.hackerrank.com/challenges/electronics-shop/problem

fun getMoneySpent(keyboards: IntArray, drives: IntArray, b: Int): Int {
    val results: MutableList<Int> = ArrayList()
    for (i in keyboards.indices) {
        for (j in drives.indices) {
            val sum = keyboards[i] + drives[j]
            if (b >= sum) results.add(sum)
        }
    }
    return if (results.isEmpty()) -1 else results.max()!!
}


internal class GetMoneySpentTest {

    @Test
    fun `test getMoneySpent`() {
        val keyboards0 = intArrayOf(3, 1)
        val drives0 = intArrayOf(5, 2, 8)
        var b = 10; var expected = 9
        var result = getMoneySpent(keyboards0, drives0, b)
        assertEquals(expected, result)
        val keyboards1 = intArrayOf(4)
        val drives1 = intArrayOf(5)
        b = 5; expected = -1
        result = getMoneySpent(keyboards1, drives1, b)
        assertEquals(expected, result)
    }

}